americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+                   positions=tick,
+                   notch=notch,
+                   **boxplot_props)
+
+    # Sets up axis formatting
+    if show_counts:
+        kwargs['counts'] = counts
+    if 'xlim' not in kwargs:
+        kwargs['xlim'] = xlim
+    if 'xticks' not in kwargs:
+        kwargs['xticks'] = ticks
+
+    _format_axis(ax, **kwargs)
+
+    return ax
+
+
+def pretty_pandas_boxplot(meta, group, cat, order=None, ax=None,
+    **boxplot_props):
+    """Creates a more attractive poxplot than pandas
+
+    Parameters
+    ----------
+    meta : pandas dataframe
+        The metadata for the variable containing a column with a continous
+        varaible, designated in `cat`, and a categorical variable, `group`
+        with categories given by `order`.
+    group : str
+        The name of a column in meta which is a categorical predictor variable
+    cat : str
+        A column in meta which contains a continous response variable
+    order : list, optional
+        The order of categories in `group`. This can be used to limit
+        the categories plotted. For instance, if there are three categories in
+        `group`: A, B and C, and you only wish to compare A and C, you can
+        list order as ['A', 'C'] to limit the categories.
+    interval : float, optional
+        The spacing between the boxplot instances on the axes
+    notch : bool, optional
+        Displays the parametric 95% confidence interval around the mean.
+    show_n : bool, optional
+        Shows the size of the groups below each plot on the x-axis
+    p_value : float, optional
+        Default is None. When supplied, the signfigance value will be displayed
+        on the plot in the upper right hand corner by default.
+    show_xgrid: bool, optional
+        Default is False. Adds vertical lines at each major x-tick.
+    show_ygrid: bool, optional
+        Default is True. Adds horizonal lines at each major y-tick.
+    title: str, optional
+        The title to be placed on the graph.
+    ylims : list
+        The limits for the y-axis.
+    ylabel : str
+        The label text for the y-axis. Every time you leave off appropriate
+        labels and units, a science grad student grading lab reports cries
+        another bitter tear into their bottle of craft beer.
+
+    Returns
+    -------
+    ax : axes
+        A matplotlib axes containing the plotted data
+
+    """
+
+    grouped = meta.groupby(group)
+
+    # Sets up the plotting order
+    if order is None:
+        order = grouped.groups.keys()
+
+    # Gets the data vectors
+    vecs = [grouped.get_group(g)[cat].values for g in order]
+
+    # Formats the axis, if not already done
+    if 'xticklabels' not in boxplot_props:
+        boxplot_props['xticklabels'] = [g.split('(')[0] for g in order]
+    if 'show_xticks' not in boxplot_props:
+        boxplot_props['show_xticks'] = False
+    if 'show_ygrid' not in boxplot_props:
+        boxplot_props['show_ygrid'] = True
```

Comment:
flagged due to the inconsistency of the logic. If `show_ygrid` is not in
your options, then the default is to show the grid. This is the exact
opposite logic to the prior if statement.

On Tue, Mar 3, 2015 at 4:22 PM, J W Debelius <notifications@github.com>
wrote:

> In americangut/diversity_analysis.py
> <https://github.com/biocore/American-Gut/pull/127#discussion_r25738404>:
>
> > +    grouped = meta.groupby(group)
> > +
> > +    # Sets up the plotting order
> > +    if order is None:
> > +        order = grouped.groups.keys()
> > +
> > +    # Gets the data vectors
> > +    vecs = [grouped.get_group(g)[cat].values for g in order]
> > +
> > +    # Formats the axis, if not already done
> > +    if 'xticklabels' not in boxplot_props:
> > +        boxplot_props['xticklabels'] = [g.split('(')[0] for g in order]
> > +    if 'show_xticks' not in boxplot_props:
> > +        boxplot_props['show_xticks'] = False
> > +    if 'show_ygrid' not in boxplot_props:
> > +        boxplot_props['show_ygrid'] = True
>
> The default for _format_axis is to not to show grids, since they're
> discouraged in almost every primer on preparing scientific figures. The lab
> seems to prefer grid lines on boxplots, and I'd make the case that they are
> help for for interpretation, so I made it default here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/American-Gut/pull/127/files#r25738404>.
>